src/support/logger/logger.h

```diff
@@ -61,4 +60,21 @@ void logger_log(Logger* logger, LogLevel level, const gchar* fileName,
                 const gchar* functionName, const gint lineNumber,
                 const gchar* format, ...);
 
+// Returns an agreed-upon start time for logging purposes, as returned by
+// logger_now_micros.
+//
+// Logger implementations should use this to get the logging "start" time.
+// This ensures consistency when switching loggers, and enables us to
+// synchronize loggers across processes.
+int64_t logger_get_global_start_time_micros();
+
+// Returns "now" according to a monotonic system clock.
+int64_t logger_now_micros();
+
+// Returns elapsed micros since agreed-upon start time.
+int64_t logger_elapsed_micros();
+
+// Set the global start time used in log messages. If this isn't called, the
+// start time will be set to the current time the first time it's accessed.
+void logger_set_global_start_time_micros(int64_t);
```

Comment:
Oops, missed this comment.

This is a matter of some debate, but in general I agree with the Google style guide that it's generally preferable to use signed integers to represent integers

https://google.github.io/styleguide/cppguide.html#Integer_Types

> Unsigned integers are good for representing bitfields and modular arithmetic. Because of historical accident, the C++ standard also uses unsigned integers to represent the size of containers - many members of the standards body believe this to be a mistake, but it is effectively impossible to fix at this point. The fact that unsigned arithmetic doesn't model the behavior of a simple integer, but is instead defined by the standard to model modular arithmetic (wrapping around on overflow/underflow), means that a significant class of bugs cannot be diagnosed by the compiler. In other cases, the defined behavior impedes optimization.

Reply:
OK. I don't feel strongly.